src/webshare.js

```diff
@@ -101,36 +163,65 @@ const webshare = {
       headers,
     );
     if (
+      !resp || 
       resp.statusCode != 200 ||
-      resp.body.children.find((el) => el.name == "status").value != "OK"
+      !resp.body || 
+      !resp.body.children ||
+      resp.body.children.find((el) => el.name == "status")?.value != "OK"
     ) {
       throw Error("Cannot log in to Webshare.cz, invalid login credentials");
     }
-    return resp.body.children.find((el) => el.name == "token").value;
+    return resp.body.children.find((el) => el.name == "token")?.value;
   },
 
   directSearch: async (query, token) => {
     return await search(query, token);
   },
 
   getById: async (id, token) => {
-    await needle("https://webshare.cz/api/file_info");
     const data = formencode({ ident: id, wst: token });
     const resp = await needle(
       "post",
       "https://webshare.cz/api/file_info/",
       data,
       { headers },
     );
+    
+    // Check if response is valid
+    if (!resp || !resp.body || !resp.body.children) {
+      console.log(`getById failed for ${id}: Invalid response structure`);
+      return null;
+    }
+    
     const children = resp.body.children;
 
-    const size = children.find((el) => el.name == "size").value;
-    const posVotes = children.find((el) => el.name == "positive_votes").value;
-    const negVotes = children.find((el) => el.name == "negative_votes").value;
-    const filename = children.find((el) => el.name == "name").value;
-    const desc = children.find((el) => el.name == "description").value;
+    // Check if status is OK
+    const status = children.find((el) => el.name == "status");
+    if (!status || status.value !== "OK") {
+      console.log(`getById failed for ${id}: Status not OK`);
+      return null;
+    }
+
+    const size = children.find((el) => el.name == "size")?.value;
+    const posVotes = children.find((el) => el.name == "positive_votes")?.value;
+    const negVotes = children.find((el) => el.name == "negative_votes")?.value;
+    const filename = children.find((el) => el.name == "name")?.value;
+    const desc = children.find((el) => el.name == "description")?.value;
     const password = children.find((el) => el.name == "password");
     const stripe = children.find((el) => el.name == "stripe")?.value;
+    // These fields may not be present in all files
+    const bitrate = children.find((el) => el.name == "bitrate")?.value;
+    const width = children.find((el) => el.name == "width")?.value;
+    const height = children.find((el) => el.name == "height")?.value;
```

Comment:
That is pretty cool we get this information! I mostly followed the official docs, but it seems the service is returning some extra fields for video, nice!